www_src/pages/make/make.jsx

```diff
@@ -98,11 +99,41 @@ var Make = React.createClass({
     }
   },
 
+  cardActionClick: function (e) {
+    dispatcher.fire('modal-switch:show', {
+      config: {
+        actions: ['Share', 'Delete'],
+        callback: (event) => {
+          if (event.label === 'Delete') {
+            api({
```

Comment:
can you `setState({loading: true})` and `false` to block user interactions here?